tests/unit/tck/protocol_test.py

```
@@ -0,0 +1,77 @@
+def test_response_formatting_success():
+    """Test formatting of a successful JSON-RPC response."""
+    resp = build_json_rpc_success_response({"ok": True}, 1)
+    assert resp["jsonrpc"] == "2.0"
+    assert "result" in resp
+
+
+def test_response_formatting_error():
+    """Test formatting of an error JSON-RPC response."""
+    class DummyError:
+        """A dummy error class for testing."""
+        def __init__(self):
```

Ruff (0.14.14)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)